src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -105,6 +107,7 @@ public override void Dispose()
                 if (_firstRejectedStreamId == -1)
                 {
                     _firstRejectedStreamId = long.MaxValue;
+                    Task.WhenAll(_streamDisposals).GetAwaiter().GetResult();
```

Comment:
Should this rather be in `CheckForShutdown`?
Is it possible to still have active requests in this moment, which would eventually add to the collection?